subprojects/core-api/src/main/java/org/gradle/api/artifacts/Configuration.java

```diff
@@ -493,27 +495,65 @@ default void fromDependencyCollector(DependencyCollector collector) {
 
     /**
      * Execute the given action before the configuration first participates in
-     * dependency resolution. A {@code Configuration} will participate in dependency resolution
-     * when:
+     * dependency resolution. Actions will be executed in the order provided.
+     * A configuration will participate in dependency resolution when:
+     *
      * <ul>
      *     <li>The {@link Configuration} itself is resolved</li>
-     *     <li>Another {@link Configuration} that extends this one is resolved</li>
-     *     <li>Another {@link Configuration} that references this one as a project dependency is resolved</li>
+     *     <li>The {@link Configuration} is published to a repository</li>
+     *     <li>The {@link Configuration} is consumed as a variant by another project</li>
+     *     <li>Another {@link Configuration} that extends this one is resolved, published, or consumed</li>
      * </ul>
+     * <p>
+     * In general, this method should be avoided in favor of other lazy APIs. However, in some cases where
+     * lazy APIs are not yet available, this method can be used to perform actions before the configuration
+     * is used.
+     * <p>
+     * Despite the method's name, callbacks registered on this method should not add dependencies to the
+     * configuration or mutate the dependencies already present on the configuration. Instead, use the
```

Comment:
Is this supposed to read more like "Despite the method's name, it should not be used to add dependencies to the configuration or mutate the dependencies already present on the configuration"? Currently it reads like "you should use this method, but not call specific things inside of your callback".